clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -38,13 +42,54 @@
     public static final String ZK_LOGIN_CONTEXT_NAME_KEY = "zookeeper.sasl.clientconfig";
 
     /**
-     * Construct a JAAS configuration object per kafka jaas configuration file
-     * @param loginContextName
-     * @param key
-     * @return JAAS configuration object
+     * Returns a JAAS Configuration object. For loginType SERVER, default Configuration
+     * is returned. For loginType CLIENT, if Jaas configuration property
+     * {@link SaslConfigs#SASL_JAAS_CONFIG} is specified, the configuration object
+     * is created by parsing the property value. Otherwise, the default Configuration
+     * is returned.
      */
-    public static String jaasConfig(String loginContextName, String key) throws IOException {
-        AppConfigurationEntry[] configurationEntries = Configuration.getConfiguration().getAppConfigurationEntry(loginContextName);
+    public static Configuration jaasConfig(LoginType loginType, Map<String, ?> configs) {
+        Password jaasConfigArgs = (Password) configs.get(SaslConfigs.SASL_JAAS_CONFIG);
+        if (jaasConfigArgs == null || loginType == LoginType.SERVER)
```

Comment:
Is there a good reason to allow `LoginType.SERVER` and `jaasConfigArgs != null`? Or would that indicate a bug somewhere (and we should throw an exception)?